apps/eth_blockchain/test/eth_blockchain/adapter_test.exs

```diff
@@ -0,0 +1,149 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule EthBlockchain.AdapterTest do
+  use ExUnit.Case
+  alias EthBlockchain.{Adapter, DumbAdapter}
+  alias Ecto.UUID
+
+  setup do
+    mock_balance = 123
+
+    Code.eval_string("""
+      defmodule MockAdapter do
+        def start_link, do: GenServer.start_link(__MODULE__, :ok, [])
+        def init(:ok), do: {:ok, nil}
+        def stop(pid), do: GenServer.stop(pid)
+
+        def handle_call({:get_balances, _address, contract_addresses, _block}, _from, reg) do
+          balances = Map.new(contract_addresses, fn ca -> {ca, #{mock_balance}} end)
+          {:reply, {:ok, balances}, reg}
+        end
+      end
+    """)
+
+    supervisor = String.to_atom("#{UUID.generate()}")
+
+    {:ok, _} =
```

Comment:
Might worth switching this to `EthBlockchainCase`. Probably make it return supervisor pid as well? (Since we need to perform the `which_children` check?)